REGR: fix eval with inplace=True to correctly update column values inplace (#47550)
pandas-dev/pandas: doc/source/whatsnew/v1.4.4.rst

Fixed regressions
~~~~~~~~~~~~~~~~~

> - Fixed regression in :meth:`DataFrame.eval` called with ``inplace=True`` had no effect (:issue:`47449`)

Reviewer: the `inplace` argument also affects the return type so can perhaps reword for clarity and to be consistent with the release note for #47327

```suggestion
- Fixed regression in :meth:`DataFrame.eval` creating a copy when updating inplace (:issue:`47449`)
```

Author: done
